wp-cli.php

```diff
@@ -354,8 +359,7 @@ public function ip_add( $args, $assoc_args ) {
 		}
 
 		// Get the new IPs.
-		$options     = $this->get_options();
-		$current_ips = empty( $options['allowed'] ) ? [] : $options['allowed'];
+		$current_ips = $this->get_current_ips();
```

Comment:
Similarly, this may also be better as the original `get_options()` call, because otherwise later on when it updates the option it adds the config ones to the option as well. This means you could end up adding one of the config-set IPs to the option but maybe that's not a terrible thing? Otherwise we will probably end up down the enhancement rabbit hole of splitting up the warning which I don't think is super necessary to get this out the door :)

Reply:
Hrm. What would the use case for adding an IP to both the option and config? I'd think we would want to block that?

Reply:
I don't know if there's a good use case for it but I think it's better than inadvertently adding _all_ of the config-set IPs to the option when you add another IP, which is what's currently happening. If you think the fix is better down below in the `array_merge()` to use the option instead (or some such) that's fine with me too, and more correct in terms of outcome.

Reply:
Absolutely agreed. I'm going to see about ensuring just the option IPs get saved, not config. Do you see anything else with this PR, or is that the last issue?

Reply:
Okay, should be resolved: d19a41dd00cc81021999e5def536247998a05a6e

Reply:
Last issue for this PR! There's a separate issue around undefined indices that I'll handle :) Thanks for your help wrapping this up!